packages/cspell/src/util/reporters.ts

```diff
@@ -0,0 +1,50 @@
+import type { CSpellReporterModule, FileSettings, ReporterSettings } from 'cspell-lib';
+import { CSpellReporter } from '@cspell/cspell-types';
+
+function mergeEmitters<T extends keyof CSpellReporter>(
+    reporters: ReadonlyArray<CSpellReporter>,
+    emitterName: T
+): CSpellReporter[T] {
+    return (...args: unknown[]) => {
+        // eslint-disable-next-line prefer-spread
+        const results = reporters.map((reporter) => reporter[emitterName].apply(reporter, args));
+        return Promise.all(results);
+    };
+}
+
+/**
+ * Mergers several cspell reporters into a single one
+ */
+export function mergeReporters(...reporters: ReadonlyArray<CSpellReporter>): CSpellReporter {
+    return {
+        issue: mergeEmitters(reporters, 'issue'),
+        info: mergeEmitters(reporters, 'info'),
+        debug: mergeEmitters(reporters, 'debug'),
+        progress: mergeEmitters(reporters, 'progress'),
+        error: mergeEmitters(reporters, 'error'),
+        result: mergeEmitters(reporters, 'result'),
+    };
+}
+
+function loadReporter(reporterSettings: ReporterSettings): CSpellReporter | undefined {
+    if (!Array.isArray(reporterSettings)) {
+        reporterSettings = [reporterSettings];
+    }
+    const [moduleName, settings] = reporterSettings;
+
+    try {
+        // eslint-disable-next-line @typescript-eslint/no-var-requires
+        const { getReporter }: CSpellReporterModule = require(moduleName);
+        return getReporter(settings);
+    } catch (e) {
+        // TODO needs better error handling?
+        console.error('Failed to load reporter', moduleName, e);
```

Comment:
Please throw an error. The `app` should catch it.

Reply:
Do we need to wrap `e` somehow here or just remove try-catch block to make original error to be thrown?

Reply:
I moved `CheckError` out of `app.ts` and into `util/errors.ts`. I think it would be better to add a new type `ApplicationError` that wraps the existing error with a new message. I think the message `Failed to load reporter: ${modulesName}` is important.